src/jdk.jfr/share/classes/jdk/jfr/internal/dcmd/ArgumentParser.java

```diff
@@ -64,10 +69,44 @@ public Map<String, Object> parse() {
             addOption(key, value);
             eatDelimiter();
         }
+        checkConflict();
         checkMandatory();
         return options;
     }
 
+    protected void checkConflict() {
```

Comment:
I think this part can be simplified, the variable names are long and there is too much state to track.

How about this?

`private void checkConflicted() {
        if (conflictedOptions.isEmpty()) {
            return;
        }

        StringBuilder sb = new StringBuilder();
        if (conflictedOptions.size() == 1) {
            sb.append("Option ");
            sb.append(conflictedOptions.remove(0));
        } else {
            sb.append("Options ");
            StringJoiner sj = new StringJoiner(", ");
            while (conflictedOptions.size() > 1) {
                sj.add(conflictedOptions.remove(0));
            }
            sb.append(sj);
            sb.append(" and " + conflictedOptions.remove(0));
        }
        sb.append(" can only be specified once.");
        throw new IllegalArgumentException(sb.toString());
    }`
    
It requires an ArrayList, but you could do:
`    if (!conflictedOptions.contains(key)) {
        conflictedOptions.add(key);
      }`
          
I don't think it's worth the complication with the JfrCommand enum to display the correct command being used in an error message.

Reply:
Thank you for your prompt reviewing.

Your idea is good for me. I will change the code in that regard as soon as I can.

I agree with the complication you mentiond. We posted about this in JBS with phrases like "with -XX:StartFlightRecording" at the end of the log message, and also mentioned jcmd. Howeber, at the existing implementation in logging the message, there is no way to identify a command such as "JFR.start". So I implemented it this way to identify the command as a last resort. 

As you say, I also think it no need to have a log message that identifies each command with the complication. Therefore, I will eliminate this complication.